### ISSUE TITLE
Make the virtualenv wheel executable for self contained boostrapping

### ISSUE THREAD

**Issue by pombreda** [pypa/virtualenv#744]
The wheel is a zip therefore it is something that is like a pyz (pep441)
It also contains vendored pip and setuptools which is mighty useful for boostrapping.

To make it a fully boostrapable self-contained archive it is missing:
 *  a simple  `__main__.py` that will launch virtualenv.py
 * a way to ensure that even if not installed/extracted yet, the vendored pip and setuptools would be used.

With these two minor enhancements, you could bootstrap a fully isolated env by just fetching the wheel and run python <wheel file> <path to your venv>

**Comment by pfmoore** [comment]
tl; dr - this is not something we particularly want to support, or are likely to add

Even though they are both zipfiles, wheels and ```pyz``` archives are different (specifically, the wheel format doesn't allow for a top-level ```__main__.py``` file). See the wheel PEP, specifcally https://www.python.org/dev/peps/pep-0427/#is-it-possible-to-import-python-code-directly-from-a-wheel-file, for a discussion of why executing directly from a wheel is viewed as a nice, but not formally supported, property.

On the specific note of the internal setuptools and pip wheels, those are embedded zipfiles within the virtualenv wheel, and using them inplace is far more complex than it maybe seems at first glance.

**Comment by pombreda** [comment]
@pfmoore fine by me... let me close this ;)

**Comment by pombreda** [comment]
a simple script can extract and do the same...
